src/globals/helpers/spacing.test.js

```diff
@@ -0,0 +1,265 @@
+/* eslint-env jest */
+
+const util = require('util')
+
+const configPaths = require('../../../config/paths.json')
+
+const outdent = require('outdent')
+const sass = require('node-sass')
+const sassRender = util.promisify(sass.render)
+
+const sassConfig = {
+  includePaths: [ configPaths.src ],
+  outputStyle: 'nested'
+}
+
+const sassBootstrap = `
+  $mq-breakpoints: (
+    my_breakpoint: 30em
+  );
+
+  $spacing-map: (
+    null: 15px,
+    my_breakpoint: 25px
+  );
+
+  @import "globals/helpers/media-queries";
+  @import "globals/tools/iff";
+  @import "globals/helpers/spacing";`
+
+describe('@mixin govuk-responsive-spacing', () => {
+  it('outputs CSS for a property based on the given spacing map', async () => {
+    const sass = `
+      ${sassBootstrap}
+
+      .foo {
+        @include govuk-responsive-spacing($spacing-map, 'margin')
+      }`
+
+    const results = await sassRender({ data: sass, ...sassConfig })
+
+    expect(results.css.toString().trim()).toBe(outdent`
+      .foo {
+        margin: 15px; }
+        @media (min-width: 30em) {
+          .foo {
+            margin: 25px; } }`)
+  })
+
+  it('outputs CSS for a property and direction based on the spacing map', async () => {
+    const sass = `
+      ${sassBootstrap}
+
+      .foo {
+        @include govuk-responsive-spacing($spacing-map, 'padding', 'top');
+      }`
+
+    const results = await sassRender({ data: sass, ...sassConfig })
+
+    expect(results.css.toString().trim()).toBe(outdent`
+      .foo {
+        padding-top: 15px; }
+        @media (min-width: 30em) {
+          .foo {
+            padding-top: 25px; } }`)
+  })
+
+  it('throws an exception when passed anything other than a map', async () => {
+    const sass = `
+      ${sassBootstrap}
+
+      .foo {
+        @include govuk-responsive-spacing(14px, 'margin')
+      }`
+
+    expect.assertions(1)
```

Comment:
May be able to do something like https://github.com/alphagov/govuk-frontend/blob/master/src/back-link/template.test.js#L20 to avoid this